nasa/ow_autonomy
# Oceanwater 950 rename actions

## Summary

This branch combines the work of the following Jiras:
- OW-950 (Rename ROS actions to unified spec)
- OW-1072 (PLEXIL interface to ROS actions)
- OW-1103 (Refactor and renamed PLEXIL plans and commands)

They were combined on this branch to expedite the completion of Release 11, and thus this is one big PR instead of three smaller sequential ones.

Specific changes and kinds of changes:
- New PLEXIL interface to StopArm and DockIngestSample actions, along with library plans and use in test plans.
- PLEXIL interfaces, library plans, and test plans for remaining actions that needed them.
- Many ROS action references were renamed and/or had parameters fixed to match the corresponding branch in `ow_simulator` (and thus the unified specification with OWLAT).
  - In PLEXIL: plan file names, plan/action references, command names
  - In the PLEXIL adapter (C++): action and PLEXIL command references, some (but not all) internal function and variable names (wanted to keep some names shorter).
 - The fledging `plans/unified` directory was folded into its parent.
 - Edits and improvements to several existing plans, including:
   - Fine tuning the scoop depth in ReferenceMission1 so that sample reaches the dock.
   - Emptying the sample doc at the end of ReferenceMission1.
   - Fleshing out `TestActions` with all the actions.
 - New Boolean PLEXIL lookups `UsingOceanWATERS` and `UsingOWLAT` for querying which testbed is in use.

## Notes
- The PLEXIL adapter is very verbose and repetitive and could use a refactor.  Wanted to employ Honeybee Robotic's https://github.com/honeybee-robotics/plexil_actionlib but didn't have time for Release 11, as it would entail a complete rewrite.
- Actions not included in unification remain in `ow_lander` and not `owl_msgs`.
- There is no Plexil test for the `ArmStop` action.  May add one later in release test.

## Test
- The workspace should be built clean with `ow_simulator` on the same branch.
- Restart the Atacama world prior to each plan run.
  - Run `ReferenceMission1` for regression.  It should run as before, but with an added `DockIngestSample` near the end.
  - Run `TestActions`.  It should run without warnings or errors.  It runs every available action as a basic proof of interface.  A more rigorous test of each action will be in the release tests.

## Known Problems

1. The following error sometimes occurs in the simulator:
```
[ERROR]: FrameTransfomer.transform failure: "world" passed to lookupTransform argument target_frame does not exist.                                                                         
[ERROR]: ArmMoveCartesian: Aborted - Trajectory planning failed   
```
